snapcraft/tests/unit/project_loader/grammar/test_to_statement.py

```diff
@@ -0,0 +1,280 @@
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'bar:amd64'}
+        }),
+        ('third else ignored', {
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [
+                ['bar'],
+                ['baz']
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'bar'}
+        }),
+        ('third else followed', {
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [
+                [{'to armhf': ['bar']}],
+                ['baz']
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'baz'}
+        }),
+        ('nested armhf', {
+            'to': 'to armhf',
+            'body': [
+                {'to armhf': ['foo']},
+                {'to i386': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'armhf',
+            'expected_packages': {'foo:armhf'}
+        }),
+        ('nested i386', {
+            'to': 'to i386',
+            'body': [
+                {'to armhf': ['foo']},
+                {'to i386': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'i386',
+            'expected_packages': {'bar:i386'}
+        }),
+        ('nested body ignored else', {
+            'to': 'to armhf',
+            'body': [
+                {'to armhf': ['foo']},
+                {'else': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'armhf',
+            'expected_packages': {'foo:armhf'}
+        }),
+        ('nested body used else', {
+            'to': 'to i386',
+            'body': [
+                {'to armhf': ['foo']},
+                {'else': ['bar']},
+            ],
+            'else_bodies': [],
+            'target_arch': 'i386',
+            'expected_packages': {'bar:i386'}
+        }),
+        ('nested else ignored else', {
+            'to': 'to i386',
+            'body': ['foo'],
+            'else_bodies': [
+                [
+                    {'to armhf': ['bar']},
+                    {'else': ['baz']},
+                ],
+            ],
+            'target_arch': 'armhf',
+            'expected_packages': {'bar:armhf'}
+        }),
+        ('nested else used else', {
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [
+                [
+                    {'to armhf': ['bar']},
+                    {'else': ['baz']},
+                ],
+            ],
+            'target_arch': 'i386',
+            'expected_packages': {'baz'}
+        }),
+    ]
+
+    @patch('platform.architecture')
+    @patch('platform.machine')
+    def test_to_statement_grammar(self, platform_machine_mock,
+                                  platform_architecture_mock):
+        platform_machine_mock.return_value = 'x86_64'
+        platform_architecture_mock.return_value = ('64bit', 'ELF')
```

Comment:
Why are we setting this, here? `to` doesn't care about the build host, right? Just the target arch, which we're setting here.